sdk/search/azure-search-documents/azure/search/documents/indexes/_internal/_search_indexer_client.py

```diff
@@ -269,21 +274,22 @@ def create_datasource(self, data_source, **kwargs):
                 :caption: Create a Data Source
         """
         kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
-        result = self._client.data_sources.create(data_source, **kwargs)
-        return result
+        packed_data_source = pack_search_indexer_data_source(data_source)
+        result = self._client.data_sources.create(packed_data_source, **kwargs)
+        return unpack_search_indexer_data_source(result)
 
     @distributed_trace
     def create_or_update_datasource(self, data_source, name=None, **kwargs):
-        # type: (SearchIndexerDataSource, Optional[str], **Any) -> Dict[str, Any]
+        # type: (SearchIndexerDataSourceConnection, Optional[str], **Any) -> SearchIndexerDataSourceConnection
```

Comment:
We should do this in a different PR prolly - but we must allow individual keywords only to update the datasource - which if present would override the datasource. (See skillsets for example)